sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,181 @@
+    ownership_list = checkpoint_store.claim_ownership(single_ownership)
+    assert ownership_list[0]['owner_id'] == 'Bill'
+
+    single_ownership = [result_ownership_list[0].copy()]
+    single_ownership[0]['etag'] =  'W/"datetime\'2021-08-02T00%3A46%3A51.7645424Z\'"'
+    single_ownership[0]['owner_id'] = 'Jack'
+    single_ownership[0]['partition_id'] = '10'
+    result_ownership = checkpoint_store.claim_ownership(single_ownership)
+    list_ownership =  checkpoint_store.list_ownership(fully_qualified_namespace, eventhub_name, consumer_group)
+    assert result_ownership[0] in list_ownership
+
+    single_ownership = [result_ownership_list[0].copy()]
+    single_ownership[0]['etag'] =  'W/"datetime\'2021-08-02T00%3A46%3A51.7645424Z\'"'
+    with pytest.raises(OwnershipLostError) as e_info:
+            checkpoint_store.claim_ownership(single_ownership)
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    ownership_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    ownership_list = checkpoint_store.list_ownership(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(ownership_list) == 0
+
+    ownership_list = []
+
+    for i in range(ownership_cnt):
+            ownership = {}
+            ownership['fully_qualified_namespace'] = fully_qualified_namespace
+            ownership['eventhub_name'] = eventhub_name
+            ownership['consumer_group'] = consumer_group
+            ownership['owner_id'] = 'owner_id'
+            ownership['partition_id'] = str(i)
+            ownership['etag'] = None
+            ownership['last_modified_time'] = None
+            ownership_list.append(ownership)
+    result_ownership_list = checkpoint_store.claim_ownership(ownership_list)
+    assert ownership_list != result_ownership_list
+    assert len(result_ownership_list) == len(ownership_list)
+    for i in range(len(ownership_list)):
+        assert ownership_list[i]['etag'] != result_ownership_list[i]['etag']
+        assert ownership_list[i]['last_modified_time'] != result_ownership_list[i]['last_modified_time']
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace, eventhub_name, consumer_group)
+    assert len(ownership_list) == ownership_cnt
+    assert len(ownership_list) == len(result_ownership_list)
+    for i in range(len(result_ownership_list)):
+        assert ownership_list[i]['etag'] == result_ownership_list[i]['etag']
+        assert ownership_list[i]['last_modified_time'] == result_ownership_list[i]['last_modified_time']
+
+def _update_and_list_checkpoint(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
```

Comment:
can be moved to `_create_checkpoint` helper method?

Reply:
so I need them for list_checkpoint and list ownership but I can add to the _create_ownership helper